table/block_based/block_based_table_reader.cc

```diff
@@ -94,7 +94,7 @@ CacheAllocationPtr CopyBufferToHeap(MemoryAllocator* allocator, Slice& buf) {
       const BlockHandle& handle, const UncompressionDict& uncompression_dict, \
       CachableEntry<T>* out_parsed_block, GetContext* get_context,            \
       BlockCacheLookupContext* lookup_context, bool for_compaction,           \
-      bool use_cache, bool async_read) const;
+      bool use_cache, bool async_read, bool skip_block_cache_lookup) const;
```

Comment:
```RetrieveBlock``` already has a ```use_cache``` argument. Can we leverage that instead of adding ```skip_block_cache_lookup```, which sounds very similar and might lead to confusion? We could modify the algorithm in ```RetrieveBlock``` slightly to make it work for this case -
```
  if (use_cache || ro.fill_cache) {
    Call MaybeReadBlockAndLoadToCache(..., /*skip_block_cache_lookup=*/ !use_cache)
  } else {
    Call ReadAndParseBlockFromFile()
  }
```

Will that work? That would simplify this PR.

Reply:
@anand1976  I tried this approach but there are some unit tests where use_cache can be false for index blocks (when cache index and filter block is false) but ro.fill_cache is set to true (default value) resulting in failure as its expects the cache to be empty.  Like one in 

https://github.com/facebook/rocksdb/blob/9a034801cead6421bcf82b506b77e3b2251f1edb/db/db_block_cache_test.cc#L172 which expects it to be 0 but it's not because use_cache is false for index blocks but add ro.fill_cache is true when DB is Reopened.